src/meili/match_route.cc

```diff
@@ -129,47 +129,39 @@ bool MergeRoute(const State& source,
 void cut_segments(const std::vector<MatchResult>& match_results,
                   int first_idx,
                   int last_idx,
-                  const std::vector<EdgeSegment>& segments,
+                  std::vector<EdgeSegment>::iterator& first_segment,
+                  const std::vector<EdgeSegment>::iterator& segments_end,
                   std::vector<EdgeSegment>& new_segments) {
-  auto first_segment = segments.begin();
   int prev_idx = first_idx;
   for (int curr_idx = first_idx + 1; curr_idx <= last_idx; ++curr_idx) {
     const MatchResult& curr_match = match_results[curr_idx];
     const MatchResult& prev_match = match_results[prev_idx];
 
-    // skip if we do not need to cut
-    if (!curr_match.is_break_point && curr_idx != last_idx) {
-      continue;
```

Comment:
this was basically the culprit: it wouldn't return/cut any edges if we're not on the very last segment for this range. but there are the `interpolated` points in between which were just skipped here